db/migrate/20190117191703_create_versioned_image_asset.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+class CreateVersionedImageAsset < ActiveRecord::Migration[5.2]
+  def change
+    create_table :versioned_image_assets do |t|
+      t.references :file_revision,
+                   foreign_key: { to_table: :versioned_image_file_revisions,
+                                  on_delete: :cascade },
+                   index: true,
+                   null: false
+
+      t.references :superseded_by,
+                    foreign_key: { to_table: :versioned_image_assets,
+                                 on_delete: :nullify },
+                    index: false,
+                    null: true
+
+      t.string :variant, null: false
+      t.string :file_url
+      t.string :state, default: "absent", null: false
+      t.timestamps
+
+      t.index :file_url, unique: true
```

Comment:
Is this a problem if we have two absent assets, where the file_url is null?

Reply:
No it's not as a unique index ignores null values (which is one of my favourite features 😁)